docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,285 @@
+
+Output parameters are not taken into account when establishing rule entry uniqnuess. For example the following rule will fail on startup because every skr can be matched by both rules:
+
+```
+hap:
+  rule: 
+    - gcp 
+    - gcp -> S # invalid entry, output attributes does not take part into uniqnuess check
+    - gcp(PR=*) # invalid entry, both can be applied to the same skr
+```
+
+The similary behaviour occurs when the rule defined the same attributes values together with `*` value.
+
+```
+hap:
+  rule: 
+    - gcp(CR=europe-west3) 
+    - gcp(PR=*, CR=europe-west3) # invalid rules, both can be applied to the same skr
+```
+
+Rule configuration must contain only unique entries in the scope of that list. 
+If rule configuration contains duplicated rule entries an error during KEB startup is returned.
+
+Rule entry priority is selected by sorting all the rules entries that apply to the request by the number of indentification attributes they contain. 
+For example, a rule with no attributes and only plan has lower priority then a rule with the same plan but a platform region attribute (`gcp` < `gcp(PR=cf-sa30)`).
+
+After sorting the entry that specifies most attributes (hence is more specific) is selected. 
+Input attributes with value `*` is not taken into account when calculating priority.
+
+The following example shows the priority of the rules from the general rule to more specific ones:
+
+```
+aws -> S # search labels: hyperscalerType: aws, shared: true
+aws(PR=cf-eu11) -> EU # search labels: hyperscalerType: aws_cf-eu11, euAccess: true
+aws(PR=cf-eu11, CR=westeu) -> EU, S # search labels: hyperscalerType: aws_cf-eu11_westeu, shared: true, euAccess: true
+```
+
+## Search Labels
+
+The HAP stores credentials for the hyperscaler accounts that have been set up in advance in Kubernetes Secrets that the Secret Bindings, that KEB searches for, point to. The SecretBindings are labeld with **hyperscalerType**, **shared** and **euAccess** labels. 
+
+The **hyperscaler-type** contains hyperscaler name and region information in the format `hyperscaler_type: <HYPERSCALER_NAME>[_<PLATFORM_REGION>][_<HYPERSCALER_REGION>]`, where both `_<PLATFORM_REGION>` and `_<HYPERSCALER_REGION>` are optional. The **hypercaler-type** label is mandatory. It is value is hardcoded and computed based on a plan in [hyperscaler_type.go](https://github.com/kyma-project/kyma-environment-broker/blob/main/common/hyperscaler/hyperscaler_type.go). This means that not all plans share their name with hyperscaler types, e.g. sap-converged-plan has `openstack` hyperscalerType and `trial` plan can have either `azure` or `aws` depending on the cluster regions. The **euAccess** and **shared** labels contain boolean values and they used to divide existing pools to secrets used by EU restricted regions and secrets shared by multiple Global Accounts. The **euAccess** and **shared** labels are optional
```

Comment:
about "converged-plan has `openstack` hyperscalerType and `trial` plan can have either `azure` or `aws` depending on the cluster regions"

1) add here a small table 
2) include azure_lite which search ro hyperscyler-type azure
3) trial "depending on the cluster regions" or provider?